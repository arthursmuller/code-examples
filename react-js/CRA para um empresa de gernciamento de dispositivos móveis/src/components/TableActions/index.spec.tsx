import { render } from '../../helper/test/test-utils';
import TableAction from "./index";


describe('Component MenuItem from Table Action', () => {
  it('should match with snapshot', () => {
    const { container } = render(<TableAction
      entityIntlLabel="entity"
      linkEdit='/edit'
      linkView='/view'
      openDestroy={() => false}
      onOpenMenu={() => false}
      onCloseMenu={() => false}
      moreItems={<div>moreItems</div>}
    />);
    expect(container).toMatchSnapshot();
  });

});
