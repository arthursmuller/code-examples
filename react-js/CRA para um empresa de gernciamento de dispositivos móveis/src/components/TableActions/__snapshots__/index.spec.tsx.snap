// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MenuItem from Table Action should match with snapshot 1`] = `
<div>
  <svg
    aria-controls="menu-list-2"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-icon chakra-menu__menu-button css-1dv2wbq"
    focusable="false"
    id="menu-button-2"
    viewBox="0 0 24 24"
  >
    <g
      id="more-horizotnal"
    >
      <circle
        cx="2"
        cy="2"
        fill="currentColor"
        id="Elipse_2"
        r="2"
        transform="translate(10 10)"
      />
      <circle
        cx="2"
        cy="2"
        fill="currentColor"
        id="Elipse_3"
        r="2"
        transform="translate(17 10)"
      />
      <circle
        cx="2"
        cy="2"
        fill="currentColor"
        id="Elipse_4"
        r="2"
        transform="translate(3 10)"
      />
    </g>
  </svg>
  <div
    class="css-r6z5ec"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-1ozmk1d"
      id="menu-list-2"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1jti93q"
        data-index="0"
        id="menu-list-2-menuitem-3"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-menu__icon-wrapper css-15v20fh"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-menu__icon css-wi95y6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M19 20H5a1 1 0 0 0 0 2h14a1 1 0 0 0 0-2z"
                />
                <path
                  d="M5 18h.09l4.17-.38a2 2 0 0 0 1.21-.57l9-9a1.92 1.92 0 0 0-.07-2.71L16.66 2.6A2 2 0 0 0 14 2.53l-9 9a2 2 0 0 0-.57 1.21L4 16.91A1 1 0 0 0 5 18zM15.27 4L18 6.73l-2 1.95L13.32 6zm-8.9 8.91L12 7.32l2.7 2.7-5.6 5.6-3 .28z"
                />
              </g>
            </svg>
          </svg>
        </span>
        <span
          style="pointer-events: none; flex: 1;"
        >
          Editar entity
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-1jti93q"
        data-index="1"
        id="menu-list-2-menuitem-4"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-menu__icon-wrapper css-15v20fh"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-menu__icon css-wi95y6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="eyes"
                transform="translate(-.159)"
              >
                <g
                  id="eye"
                >
                  <g
                    id="eye-2"
                  >
                    <path
                      d="M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zM12.22 17c-4.31.1-7.12-3.59-8-5 1-1.61 3.61-4.9 7.61-5 4.29-.11 7.11 3.59 8 5-1.03 1.61-3.61 4.9-7.61 5z"
                      fill="#a0a0a5"
                      id="Caminho_78"
                    />
                    <path
                      d="M12 8.5a3.5 3.5 0 1 0 3.5 3.5A3.5 3.5 0 0 0 12 8.5zm0 5a1.5 1.5 0 1 1 1.5-1.5 1.5 1.5 0 0 1-1.5 1.5z"
                      fill="#a0a0a5"
                      id="Caminho_79"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </svg>
        </span>
        <span
          style="pointer-events: none; flex: 1;"
        >
          Visualizar entity
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-jrqv77"
        data-index="2"
        id="menu-list-2-menuitem-5"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-menu__icon-wrapper css-15v20fh"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-menu__icon css-wi95y6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <g>
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 4.33c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4zM18 19a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V8h12z"
                      transform="translate(-776 -125) translate(776 121) translate(0 4)"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </svg>
        </span>
        <span
          style="pointer-events: none; flex: 1;"
        >
          Remover entity
        </span>
      </button>
      <div>
        moreItems
      </div>
    </div>
  </div>
</div>
`;
